corehq/apps/accounting/tests/test_autopay_with_api.py

```diff
@@ -0,0 +1,145 @@
+from django.core import mail
+
+from django_prbac.models import Role
+import stripe
+
+from dimagi.utils.dates import add_months_to_date
+
+from corehq.apps.accounting import tasks, utils
+from corehq.apps.accounting.models import (
+    Invoice,
+    PaymentRecord,
+    SoftwarePlan,
+    SoftwarePlanVersion,
+    SoftwareProductRate,
+    StripePaymentMethod,
+)
+from corehq.apps.accounting.payment_handlers import (
+    AutoPayInvoicePaymentHandler,
+)
+from corehq.apps.accounting.tests import generator
+from corehq.apps.accounting.tests.test_invoicing import BaseInvoiceTestCase
+from django.db import transaction
+
+
+class TestBillingAutoPay(BaseInvoiceTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestBillingAutoPay, cls).setUpClass()
+        cls._generate_autopayable_entities()
+        cls._generate_non_autopayable_entities()
+        cls._generate_invoices()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.non_autopay_domain.delete()
```

Comment:
Suggestion: use `cls.addClassCleanup(cls.non_autopay_domain.delete)` where `non_autopay_domain` is created and remove this `tearDownClass` method.

Reply:
Thank you! I plan to use `addCleanup` and `addClassCleanup` in my future test writing, somehow I missed this one. Fixed in 03feb8b026def36e2919dc9a23b595b9233d5a7f